Sources/BraveWallet/Crypto/Portfolio/AddCustomAssetView.swift

```diff
@@ -239,17 +310,57 @@ struct AddCustomAssetView: View {
   private func resignFirstResponder() {
     UIApplication.shared.sendAction(#selector(UIResponder.resignFirstResponder), to: nil, from: nil, for: nil)
   }
+  
+  private func clearInput() {
+    nameInput = ""
+    addressInput = ""
+    symbolInput = ""
+    decimalsInput = ""
+    tokenId = ""
+    logo = ""
+    coingeckoId = ""
+    networkSelectionStore.networkSelectionInForm = nil
+  }
 
   private func addCustomToken() {
-    userAssetStore.addUserAsset(
-      address: addressInput,
-      name: nameInput,
-      symbol: symbolInput,
-      decimals: Int(decimalsInput) ?? Int((networkSelectionStore.networkSelectionInForm?.decimals ?? 18)),
-      network: networkSelectionStore.networkSelectionInForm ?? networkStore.selectedChain,
-      logo: logo,
-      coingeckoId: coingeckoId
-    ) { [self] success in
+    let network = networkSelectionStore.networkSelectionInForm ?? networkStore.selectedChain
+    let token: BraveWallet.BlockchainToken
+    if tokenType == TokenType.token.rawValue {
+      token = BraveWallet.BlockchainToken(
+        contractAddress: addressInput,
+        name: nameInput,
+        logo: logo,
+        isErc20: network.coin != .sol,
+        isErc721: false,
+        symbol: symbolInput,
+        decimals: Int32(decimalsInput) ?? Int32((networkSelectionStore.networkSelectionInForm?.decimals ?? 18)),
+        visible: true,
+        tokenId: "",
+        coingeckoId: coingeckoId,
+        chainId: network.chainId,
+        coin: network.coin
+      )
+    } else {
+      var tokenIdTHex = ""
```

Comment:
Typo in variable name I think? `tokenIdTHex` -> `tokenIdToHex`

Reply:
fixed in [299d530](https://github.com/brave/brave-ios/pull/6360/commits/299d530805abcc2e35a9cce53897e5f42c774c62)